aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2152,6 +2155,25 @@ impl VMValidator for AptosVM {
             }
         }
 
+        if !self.features.is_zkid_enabled() || !self.features.is_open_id_signature_enabled() {
+            if let Ok(authenticators) = aptos_types::zkid::get_zkid_authenticators(&transaction) {
```

Comment:
why not `map_err(...)?`? Can be easier to read than if-else

Reply:
Sorry, @heliuchuan and I don't follow: can you replace fully with your proposed code?

Reply:
Thought that `VMValidatorResult` is an actual result for a second. Please ignore 😁